core/middlewares/with-routes.ts

```diff
@@ -223,6 +223,23 @@ const clearLocaleFromPath = (path: string, locale: string) => {
   return path;
 };
 
+function normalizeForCompare(url: URL): string {
+  const trailingSlashDisabled = process.env.TRAILING_SLASH === 'false';
+
+  if (trailingSlashDisabled && url.pathname !== '/' && url.pathname.endsWith('/')) {
+    return `${url.pathname.replace(/\/+$/, '')}${url.search}`;
+  }
+
+  if (!trailingSlashDisabled && !url.pathname.endsWith('/')) {
+    return `${url.pathname}/${url.search}`;
+  }
+
+  return `${url.pathname}${url.search}`;
+}
```

Comment:
🍹 Is this normalization necessary? I would've thought that comparing the request path to the destination path would've been enough to detect an infinite redirect. Is it because we handle the trailing slash ourselves in a middleware that's later in the chain (or perhaps before)?

Reply:
@migueloller Yes, it's because of the `TRAILING_SLASH` environment variable. In the getRoutes query, the destination path might be `/my-path/` (with trailing slash), but your location might be `/my-path`.

Without this, `/my-path` would redirect to `/my-path/` and then NextJS (TRAILING_SLASH=false) will redirect to `/my-path`, then get redirected to `/my-path/` and so-on. That's why this issue was occurring